lib/ClangImporter/ImportName.cpp

```diff
@@ -1435,9 +1461,24 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
     auto enumInfo = getEnumInfo(enumDecl);
 
     StringRef removePrefix = enumInfo.getConstantNamePrefix();
-    if (!removePrefix.empty() && baseName.startswith(removePrefix)) {
-      baseName = baseName.substr(removePrefix.size());
-      strippedPrefix = true;
+    if (!removePrefix.empty()) {
+      if (baseName.startswith(removePrefix)) {
+        baseName = baseName.substr(removePrefix.size());
+        strippedPrefix = true;
+      } else if (givenName) {
+        // Calculate the new prefix.
+        // What if the preferred name causes longer prefix?
```

Comment:
This isn't really correct, but I guess you can put that in later. (The prefix computation takes all members into account, as you note.)